src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java

```diff
@@ -94,24 +92,26 @@ protected void cleanResultsAndPublishCancel(UUID resultUuid, String receiver) {
         }
     }
 
-    private boolean cancelAsync(CancelContext cancelContext) {
+    private void cancelAsync(CancelContext cancelContext) {
         lockRunAndCancel.lock();
-        boolean isCanceled = false;
         try {
             cancelComputationRequests.put(cancelContext.resultUuid(), cancelContext);
 
             // find the completableFuture associated with result uuid
             CompletableFuture<R> future = futures.get(cancelContext.resultUuid());
-            if (future != null) {
-                isCanceled = future.cancel(true);  // cancel computation in progress
-                if (isCanceled) {
+            if (future == null) {
+                cleanResultsAndPublishCancel(cancelContext.resultUuid(), cancelContext.receiver());
```

Comment:
this will be run on all but one server ?

Reply:
it looks like it's already the case after this method returns...?

Reply:
before we did publishCancelFail but now we do cleanResults and publishStop

Reply:
removed